src/containers/projects/new/step-2.svelte

```diff
@@ -1,123 +1,177 @@
     delete newActivity.id;
-    newActivity._type = ActivityTypes.NEW;
-    activityList.push(newActivity);
+    createActivity(newActivity);
+  }
+
+  async function createActivity(activity) {
+    try {
+      ({
+        id: activity.id,
+        project: activity.project,
+      } = await api.json(api.post(`/projects/${$project.id}/activities`, {
+        data: prepareForBackend(activity),
+        csrfToken: $session.account.csrf_token,
+      })));
+      activity._editing = true;
+      $project.activities.push(activity);
+      project = project;
+    } catch (e) {
+      showSnackbar({
+        props: { text: 'Couldn\'t create the activity, try reloading the page' },
+      });
+      console.error(e);
+    }
     scrollToLast = true;
-    activityList = activityList;
   }
 
-  function deleteActivity({ detail: activity }) {
-    activityList = activityList.filter(
-      act => act.name !== activity.name || (act._type !== ActivityTypes.DISPLAY && act._type !== ActivityTypes.TEMPLATE),
-    );
-    dispatch('delete-activity', activity.id || activity.name);
-    if (activityList.length === 0) {
-      const activity = getBlankActivity();
-      activity._type = ActivityTypes.NEW;
-      activityList.push(activity);
+  async function deleteActivity({ detail: activity }) {
+    try {
+      await api.json(api.del(
+        `/projects/${$project.id}/activities/${activity.id}`,
+        { csrfToken: $session.account.csrf_token },
+      ));
+      $project.activities = $project.activities.filter(act => act.id !== activity.id);
+      if ($project.activities.find(act => !act.internal) == null) {
+        createActivity(getBlankActivity());
+      }
+
+      showSnackbar({
+        props: {
+          text: 'Activity deleted',
+          action: { text: 'undo', callback() { createActivity(activity); } },
+        },
+      });
+    } catch (e) {
+      showSnackbar({
+        props: { text: 'Couldn\'t delete the activity, try reloading the page' },
+      });
+      console.error(e);
+    }
+  }
+
+  function propagateDates({ detail }) {
+    for (let activity of $project.activities) {
+      if (activity.internal) {
+        continue;
+      }
+
+      if (activity.timeframe == null
+          || activity.timeframe.start == null && activity.timeframe.end == null) {
+        api.json(api.patch(`/projects/${activity.project}/activities/${activity.id}`, {
+          data: prepareForBackend({ timeframe: detail.value }),
+          csrfToken: $session.account.csrf_token,
+        })).then(() => {
+          activity.timeframe = {
+            start: new Date(detail.value.start.valueOf()),
+            end: new Date(detail.value.end.valueOf()),
+          };
+        });
+      }
+    }
+  }
+
+  function propagateQuestions({ detail }) {
+    for (let activity of $project.activities) {
+      if (activity.internal) {
+        continue;
+      }
+
+      if (arraysEqual(activity.feedback_questions, defaultFeedbackQuestions)) {
```

Comment:
```suggestion
      if (!arraysEqual(activity.feedback_questions, defaultFeedbackQuestions)) continue;
```

Reply:
Also not worth it. The indented block isn't that large, and inverting the condition affects understanding